client/src/app.jsx

```diff
@@ -1,77 +1,126 @@
 import React from 'react';
 import axios from 'axios';
+import URL from 'url-parse';
 
-import processReviewsArray from './helpers/apphelpers';
+import { filterReviews } from './helpers/apphelpers';
 
 import ContainerAverageStars from './components/ContainerAverageStars';
 import ContainerReviews from './components/ContainerReviews';
 
 import {
-  OuterWrapper,
-  InnerWrapper,
+  AppOuterWrapper,
+  AppInnerWrapper,
 } from './App.styles';
 
+import { BasicTextFormatSpan } from './components/styles/MasterStyles.styles';
+
 class App extends React.Component {
   constructor(props) {
     super(props);
 
+    this.getListingId = this.getListingId.bind(this);
+
     // API calls
     this.getListingAverageStars = this.getListingAverageStars.bind(this);
     this.getListingReviews = this.getListingReviews.bind(this);
 
+    // Methods
+    this.handleSearchChange = this.handleSearchChange.bind(this);
+    this.handleSearchSubmit = this.handleSearchSubmit.bind(this);
+
     this.state = {
       averageStarsObj: {},
-      reviews: [],
+      allReviews: [],
+      renderedReviews: [],
+      currentSearch: '',
+      err404: false,
     };
   }
 
   componentDidMount() {
-    // Using parameter 1 for now, just for testing
-    this.getListingReviews(14);
-    this.getListingAverageStars(14);
+    this.getListingReviews(this.getListingId());
```

Comment:
You are calculating the id twice here, you can refactor this to only run getListingId() once or maybe even include it in the componentDidMount() and get rid of the function entirely.

Reply:
Great call!  I will make that change.